publication-commons/src/main/java/no/unit/nva/publication/service/impl/PublishingService.java

```diff
@@ -0,0 +1,169 @@
+package no.unit.nva.publication.service.impl;
+
+import java.net.URI;
+import java.util.Optional;
+import no.unit.nva.clients.ChannelClaimDto;
+import no.unit.nva.clients.IdentityServiceClient;
+import no.unit.nva.identifiers.SortableIdentifier;
+import no.unit.nva.model.EntityDescription;
+import no.unit.nva.model.PublicationOperation;
+import no.unit.nva.model.Reference;
+import no.unit.nva.model.contexttypes.Book;
+import no.unit.nva.model.contexttypes.Degree;
+import no.unit.nva.model.contexttypes.Publisher;
+import no.unit.nva.publication.model.business.FilesApprovalThesis;
+import no.unit.nva.publication.model.business.PublishingRequestCase;
+import no.unit.nva.publication.model.business.PublishingWorkflow;
+import no.unit.nva.publication.model.business.Resource;
+import no.unit.nva.publication.model.business.UserInstance;
+import no.unit.nva.publication.permissions.publication.PublicationPermissions;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+import nva.commons.apigateway.exceptions.BadGatewayException;
+import nva.commons.apigateway.exceptions.ForbiddenException;
+import nva.commons.apigateway.exceptions.NotFoundException;
+import nva.commons.core.Environment;
+import nva.commons.core.JacocoGenerated;
+import nva.commons.core.StringUtils;
+import nva.commons.core.paths.UriWrapper;
+
+public class PublishingService {
+
+    private static final String API_HOST = new Environment().readEnv("API_HOST");
+    protected static final String CUSTOMER = "customer";
+    protected static final String CHANNEL_CLAIM = "channel-claim";
+    protected static final String EVERYONE = "Everyone";
+    private final ResourceService resourceService;
+    private final TicketService ticketService;
+    private final IdentityServiceClient identityServiceClient;
+
+    public PublishingService(ResourceService resourceService, TicketService ticketService,
+                             IdentityServiceClient identityServiceClient) {
+        this.resourceService = resourceService;
+        this.ticketService = ticketService;
+        this.identityServiceClient = identityServiceClient;
+    }
+
+    @JacocoGenerated
+    public static PublishingService defaultService() {
+        return new PublishingService(ResourceService.defaultService(), TicketService.defaultService(),
+                                     IdentityServiceClient.prepare());
+    }
+
+    public void publishResource(SortableIdentifier resourceIdentifier, UserInstance userInstance)
+        throws ApiGatewayException {
+        var resource = getResource(resourceIdentifier);
+        validatePermissions(resource, userInstance);
+
+        publishPublication(userInstance, resource);
+
+        if (!resource.getPendingFiles().isEmpty()) {
+            publishResourceWithPendingFiles(userInstance, resource);
+        }
```

Comment:
Rules for publishing and ticket creation are described here:

https://sikt.atlassian.net/browse/NP-48980